feat(docs): update backport documentation (#1989)
spinnaker/spinnaker.github.io: community/contributing/releasing/index.md

In order to be considered safe to merge into a release branch, your patch must:

* Fix a documented regression. This means that the currently broken

>   functionality must have worked as expected in the version(s) of Spinnaker

Reviewer: ```suggestion
  functionality must have worked as expected in a previous version of Spinnaker.
```

>   functionality must have worked as expected in the version(s) of Spinnaker

Reviewer 2: Does the version need to be one that is still supported and if so how can a user figure out which versions are still supported? Sorry if this is a silly question.

Reviewer 3: No question is a silly questions here!

A user can view it on the releases page: https://spinnaker.io/community/releases/versions/
<kbd>![image](https://user-images.githubusercontent.com/3469532/88580337-50a83600-d000-11ea-8e04-cfc0a03c372a.png)</kbd>

or from halyard by doing a `hal version list`
<kbd>![image](https://user-images.githubusercontent.com/3469532/88580490-82b99800-d000-11ea-8a8d-baefc92f2b30.png)</kbd>

Author: 👍  Updated to include link to supported versions!

>   to which you want to backport a fix. If the regression is not already

Reviewer: ```suggestion
  If the regression is not already
```

  documented in a GitHub issue, please create one. Describe the difference

>   between the expected and observed behavior, and include links to the commit(s)

Reviewer: ```suggestion
  between the expected and observed behavior, and include links to the commit
```

Reviewer: Does it happen often that a bug is introduced by more than one commit?

Author: Very good point; no!

>   that introduced the regression. Indicate to which release(s) you would like

Reviewer: ```suggestion
  that introduced the regression. Indicate to which releases you would like
```

Author: I think we should keep this plural; contributors frequently want to backport a patch to all supported versions.

Author: Oh whoops sorry I can't read, read that as suggesting singular "release."

>   your fix to be considered a candidate for backporting.

Reviewer: ```suggestion
  your fix to be backported.
```

are built from the `release-1.16.x` release branch. To get your patch into 1.16,
it must be cherry-picked onto that release branch.

> After creating a pull request that you would like to be backported to a release

Reviewer: ```suggestion
After you've created a pull request for a fix that you want backported to a release
```

Author: Well, I'm not sure this makes more sense, since the contributor-made PR itself isn't backporting the fix (the Mergify-generated PR will later, if it is approved by the release manager).

Reviewer: Better?

Author: Love it! 🙏
